AsyncQueue.test.ts

```diff
@@ -0,0 +1,114 @@
+import { AsyncQueue } from "./AsyncQueue";
+import { setTimeout } from "timers/promises";
+
+const uploadFile = (fileName: string, file: string) =>
+  new Promise<void>(async (resolve) => {
+    logWrapper(`Started uploading: ${fileName}, ${file}`);
+    await setTimeout(1000);
+    logWrapper(`Done uploading: ${fileName}`);
+    resolve();
+  });
+
+const cleanUp = async (
+  queue: AsyncQueue<any>,
+  timeout = 1000,
+  maxTries = 5
+) => {
+  queue.stop();
+  while (queue.getQueueSize() > 0 && maxTries > 0) {
+    await setTimeout(timeout);
+    maxTries--;
+  }
+};
+
+let logWrapper = (message: string) => {
+  console.log(message);
+};
+
+describe("Testing AsyncQueue", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test("Testing AsyncQueue with one file", async () => {
+    // ARRANGE
+    logWrapper = jest.fn();
+    const fileUploadQueue = new AsyncQueue<string>(uploadFile);
+
+    // ACT
+    fileUploadQueue.add("1", "file1");
+    await cleanUp(fileUploadQueue);
+
+    // ASSERT
+    expect(logWrapper).toHaveBeenCalledTimes(2);
+    expect(logWrapper).toHaveBeenCalledWith("Started uploading: 1, file1");
```

Comment:
Please be more specific. it is related to all the assertions after this one.
```suggestion
    expect(logWrapper).toHaveBeenNthCalledWith(1, "Started uploading: 1, file1");
```